### PR TITLE
Add extensions definition to express-graphql options

### PR DESCRIPTION
**Pull request by zya**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/graphql/express-graphql#options

### CODE REVIEW

**Comment by zya** [comment]
@isman-usoh @nitintutlani @hubel

**Review by people-ware — APPROVED** [review]
The type of the extensions function is correct though incomplete

File: types/express-graphql/index.d.ts [hunk 1]
```diff
@@ -48,6 +49,13 @@ declare namespace graphqlHTTP {
          * A boolean to optionally enable GraphiQL mode.
          */
         graphiql?: boolean,
+
+        /**
+         * An optional function for adding additional metadata to the GraphQL response as a key-value object.
+         * The result will be added to "extensions" field in the resulting JSON.
+         */
+        extensions?: (document: object, variables: object, operationName: any, raw: object) => any;
```
[the remark this answers is not in the document]
Comment (zya) [reply types/express-graphql/index.d.ts]: On the first point. But you can return null in the extensions function. That is why I have put `any` as the return type. 

On the second, I'll try to add the async support to.

**Comment by zya** [comment]
@Hubel I added the async definition as well.

**Comment by Hubel** [comment]
Great. That will prevent confusion if someone needs to resolve the extensions asynchronously.

**Pull request merged**